quinn-proto/src/tests.rs

```diff
@@ -923,15 +925,20 @@ fn idle_timeout() {
     let mut pair = Pair::default();
     let (client_ch, server_ch) = pair.connect();
     pair.client.ping(client_ch);
+    let mut advanced = true;
     while !pair.client.connection(client_ch).is_closed()
         || !pair.server.connection(server_ch).is_closed()
     {
         if !pair.step() {
-            pair.time = cmp::min(pair.client.next_wakeup(), pair.server.next_wakeup());
+            if let Some(time) = min_opt(pair.client.next_wakeup(), pair.server.next_wakeup()) {
+                pair.time = time;
+            } else {
+                advanced = false;
+            }
         }
         pair.client.inbound.clear(); // Simulate total S->C packet loss
     }
-    assert!(pair.time != u64::max_value());
+    assert!(advanced);
```

Comment:
Maybe assert that the elapsed time is >= the configured idle timeout instead?

Reply:
Now asserting that the `next` coming out of the `min_opt(next wakeups)` is `Some`.